src/NServiceBus.RavenDB/Outbox/OutboxPersister.cs

```diff
@@ -68,7 +70,8 @@ public void SetAsDispatched(string messageId)
             {
                 session.Advanced.UseOptimisticConcurrency = true;
                 session.Advanced.AllowNonAuthoritativeInformation = false;
-                var outboxMessage = session.Load<OutboxRecord>(GetOutboxRecordId(messageId));
+                var outboxMessage = session.Load<OutboxRecord>(new[] { GetOutboxRecordId(messageId), GetOutboxRecordIdWithoutEndpointName(messageId) }).FirstOrDefault(o => o != null);
```

Comment:
`FirstOrDefault()` so can we have both? Or would `SingleOrDefault()` be the better choice?